contributors/design-proposals/bootstrap-discovery.md

```diff
@@ -100,7 +104,7 @@ The user experience for joining a cluster would be something like:
 kubeadm join --token=ae23dc.faddc87f5a5ab458 <address>
 ```
 
-**Note:** This is logically a different use of the token from TLS bootstrap.  We harmonize these usages and allow the same token to play double duty.
+**Note:** This is logically a different use of the token used for authentication for TLS bootstrap.  We harmonize these usages and allow the same token to play double duty.
 
 #### Implementation Flow
 
```

Comment:
(Commenting here as github won't let me comment on the other lines)

>  * Implementation note: the API server doesn't have to expose a new and special insecure HTTP endpoint.
> * `kubeadm` requests a ConfigMap containing the kubeconfig file defined above.
>   * This ConfigMap exists at a well known URL: `https://<server>/api/v1/namespaces/kube-public/configmaps/cluster-info`
>   * This ConfigMap is really public.  Users don't need to authenticate to read this ConfigMap.  In fact, the client MUST NOT use a bearer token here as we don't trust this endpoint yet.

This is not a good idea.  Unauthenticated access to the same endpoint _always_ evolves into a security vulnerability.

I presume this is not enabled by default, or that it can be turned off?   Have the k8s security folk signed off on this?